Dockerfile

```diff
@@ -12,6 +12,10 @@ COPY --from=builder /go/src/github.com/openshift/machine-config-operator/instroo
 RUN cd / && tar xf /tmp/instroot.tar && rm -f /tmp/instroot.tar
 COPY install /manifests
 
+RUN dnf -y update && dnf -y reinstall shadow-utils && \
+dnf -y install skopeo && dnf -y install openshift-clients && \
```

Comment:
It's a bit unfortunate to add a whole new copy of skopeo and `oc` into the image, because we have them right there on the host too...
(And for that matter, it's actually useful to validate the proxy configuration *from the host network namespace* since that's where most image pulling will be happening)
Tricky to deal with without making the MCC privileged enough for host mounts though.  But OTOH, the MCC really is privileged in a cluster sense anyways, so making it a privileged container (at least enough for host mounts) isn't really adding any new attack surface.

Reply:
@cgwalters I agree with the host network part, you made a great point here, that may be something worth considering.

However, regarding including the binaries, that's a usual burden that we are already paying a lot of times in a lot of components, nothing that should be surprising to us if we have OCP4 consisting of a number of clusteroperators which deploy a number of operands, almost all of them running inside containers. 

Making the MCC deployment require access to the host and require the host to always have these binaries available, even when not crazy in practice, goes much against the spirit of having every component in a container so it is, well, self-contained.

A possible way of improvement here would be to make MCO image derive from the `tools` image shipped in the release instead of `base`, because the `tools` image includes the correct version of `oc` already. Other images that require `oc` already do it and benefit from image layer de-duplication in what regards the oc client.

Reply:
> A possible way of improvement here would be to make MCO image derive from the tools image shipped in the release instead of base, because the tools image includes the correct version of oc already.

Yeah, that'd help, but doesn't get us out of also shipping skopeo, which today also vendors large parts of the container runtime again.

Hmm, do we actually need to use skopeo vs just forking `oc`?

Reply:
Yes, we do. `oc` is only used to find out which image must be checked (CNO image). Once found, the check is done with `skopeo` (`oc` doesn't provide a way to do it).

Reply:
Wouldn't just running e.g. `oc image info` be sufficient?

Reply:
mmm, as long as executing this successfully guarantees a successful pull, it would.

Reply:
`oc` today vendors the docker Go library for interacting with registries, whereas skopeo uses the github.com/containers/image bits.  But ultimately...I can't imagine a case where one worked but not the other.

Today `oc`'s fetching is kind of load-bearing because it's where we have e.g. `oc image mirror` etc that people use for disconnected.

I can't imagine a case where `oc` succeeds but `skopeo` (and/or podman/cri-o) would fail with respect to the proxy.

Reply:
I agree, we can just run `oc image info`